src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2379,28 +2379,15 @@ namespace cryptonote
         KV_SERIALIZE_N(data.start_height, "start_height")
         KV_SERIALIZE(binary)
         KV_SERIALIZE(compress)
-        if (this_ref.binary)
+        if (this_ref.binary && is_store && this_ref.compress)
```

Comment:
Are the following two macro expansions equivalent in effect? If not, this change appears to break backwards compatibility:

> KV_SERIALIZE_CONTAINER_POD_AS_BLOB_N(data.distribution, "distribution")

> KV_SERIALIZE_N(data.distribution, "distribution")


Also, if the proposed simplification is, in fact, backwards compatible, it can be expressed as follows:

```
if (this_ref.binary && this_ref.compress)
{
  if (is_store) 
  {
    const_cast<std::string&>(this_ref.compressed_data) = compress_integer_array(this_ref.data.distribution);
    KV_SERIALIZE(compressed_data)  
  }
  else
  {
    KV_SERIALIZE(compressed_data)
    const_cast<std::vector<uint64_t>&>(this_ref.data.distribution) = decompress_integer_array<uint64_t>(this_ref.compressed_data);
  }
}
else
{
  KV_SERIALIZE_N(data.distribution, "distribution")
}
```

The proposed change would have the effect of clarifying the branching strategy: binary + compressed follow one path (with load and store being distinct), everything else follows a different path.

Reply:
So `KV_SERIALIZE_CONTAINER_POD_AS_BLOB_N(data.distribution, "distribution")` isn't portable between little / big endian systems. We work around it by forcing all client requests to set "compress" to true so the portable method gets called instead.

Reply:
You probably shouldn't consider "endianness of response depends on endianness of host" a behavior to preserve, as the "remote" in RPC implies the behavior should be uniform regardless of the host. If you just forced everything to little endian, very few clients would be affected and there'd be a clear specification for future callers.

Reply:
> You probably shouldn't consider "endianness of response depends on endianness of host" a behavior to preserve, as the "remote" in RPC implies the behavior should be uniform regardless of the host.

`portable_storage` encodes everything as little endian for the network transfer and then encodes it with / without bytes swap depending on receiver machine.

> If you just forced everything to little endian, very few clients would be affected and there'd be a clear specification for future callers.

This code should support all cases now. Could you rephrase what exactly you are suggesting and why?

Reply:
I meant that, based on the comments and current code, it seems you intend to preserve the old key and behavior until some future hardfork, while introducing a new key with portable semantics, whereas the path of least resistance would perhaps involve fixing the prior serialization.